Orange/widgets/data/tests/test_owpythonscript.py

```diff
@@ -0,0 +1,123 @@
+# Test methods with long descriptive names can omit docstrings
+# pylint: disable=missing-docstring
+from Orange.data import Table
+from Orange.classification import LogisticRegressionLearner
+from Orange.widgets.data.owpythonscript import OWPythonScript
+from Orange.widgets.tests.base import WidgetTest
+
+
+class TestOWPythonScript(WidgetTest):
+    def setUp(self):
+        self.widget = self.create_widget(OWPythonScript)
+        self.iris = Table("iris")
+        self.learner = LogisticRegressionLearner()
+        self.model = self.learner(self.iris)
+
+    def test_in_data(self):
+        """Check widget's data with data on the input"""
+        self.assertEqual(self.widget.in_data, None)
+        self.send_signal("in_data", self.iris)
+        self.assertEqual(self.widget.in_data, self.iris)
+
+    def test_in_data_disconnect(self):
+        """Check widget's data after disconnecting data on the input"""
+        self.send_signal("in_data", self.iris)
+        self.assertEqual(self.widget.in_data, self.iris)
+        self.send_signal("in_data", None)
+        self.assertEqual(self.widget.in_data, None)
+
+    def test_in_learner(self):
+        """Check widget's learner with learner on the input"""
+        self.assertEqual(self.widget.in_learner, None)
+        self.send_signal("in_learner", self.learner)
+        self.assertEqual(self.widget.in_learner, self.learner)
+
+    def test_in_learner_disconnect(self):
+        """Check widget's learner after disconnecting learner on the input"""
+        self.send_signal("in_learner", self.learner)
+        self.assertEqual(self.widget.in_learner, self.learner)
+        self.send_signal("in_learner", None)
+        self.assertEqual(self.widget.in_learner, None)
+
+    def test_in_classifier(self):
+        """Check widget's model with model on the input"""
+        self.assertEqual(self.widget.in_classifier, None)
+        self.send_signal("in_classifier", self.model)
+        self.assertEqual(self.widget.in_classifier, self.model)
+
+    def test_in_classifier_disconnect(self):
+        """Check widget's classifier after disconnecting classifier on the input"""
+        self.send_signal("in_classifier", self.model)
+        self.assertEqual(self.widget.in_classifier, self.model)
+        self.send_signal("in_classifier", None)
+        self.assertEqual(self.widget.in_classifier, None)
```

Comment:
`*_disconnect` tests seem to share at least two lines with their connect counterparts.

I'd personally prefer these tests more condensed, like:
```py
def test_inputs(self):
    for input in ('in_data', 'in_learner', ...):
        ... # full connect and disconnect scenario

def test_outputs(self):
    ...
```
The basic structure of all these tests looks compatible so there is no reason to quintuplicate.